run_test.sh

```diff
@@ -36,4 +46,4 @@ cp travis-tests.ini ../../rt-WarpX
 
 # Run the tests
 cd ../../regression_testing/
-python regtest.py ../rt-WarpX/travis-tests.ini --no_update all --source_git_hash=$WARPX_TEST_COMMIT
+python regtest.py ../rt-WarpX/travis-tests.ini --no_update all --source_git_hash=${WARPX_TEST_COMMIT}
```

Comment:
todo: `mktemp` seams to create directory with a `.` in its name, which this line now chokes on (OSX).

Reply:
@RemiLehe although there is still as bug in the regression test script that we looked on, I also changed the name of the temporary directory now on OSX so it will not contain a `.` anymore. Therefore, we should be good to merge this PR already :)